ingest/scripts/split-cdsCoverage-columns.py

```diff
@@ -76,17 +81,22 @@ def main():
         if args.cdsCoverage not in reader.fieldnames:
             raise ValueError(f"The column '{args.cdsCoverage}' does not exist in the metadata '{args.metadata}'")
 
-        # Create new gene_coverage field names
-        new_fields = [f"{gene}_coverage" for gene in genes]
-        fieldnames = reader.fieldnames + new_fields
-        writer = csv.DictWriter(outfile, fieldnames=fieldnames, delimiter="\t")
-        writer.writeheader()
+        # Defer defining the writer until we process the first row, to avoid fieldname mismatch
+        writer = None
 
         # Parse gene_coverage values
         for row in reader:
             field_val = row.get(args.cdsCoverage, "")
-            gene_coverage_dict = parse_coverage_field(field_val, genes, args.round)
+            gene_coverage_dict = parse_coverage_field(field_val, genes, args.round, args.suffix)
             row.update(gene_coverage_dict)
+
+            # Only initialize writer and print header on first iteration
+            if writer is None:
+                fieldnames = list(reader.fieldnames) + list(gene_coverage_dict)
+                writer = csv.DictWriter(outfile, fieldnames=fieldnames, delimiter="\t")
```

Comment:
This runs into the risk of duplicate field names, e.g. when the metadata already has a `<gene>_coverage` column.  Since the `row` is updated with the `gene_coverage_dict`, the keys are guaranteed to be unique and can be used directly:

```suggestion
                writer = csv.DictWriter(outfile, fieldnames=list(row), delimiter="\t")
```